sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java

```diff
@@ -150,9 +150,7 @@ private Flux<FeedResponse<T>>  wrapWithTracingIfEnabled(CosmosPagedFluxOptions p
         FeedOperationState state = pagedFluxOptions.getFeedOperationState();
         DiagnosticsProvider tracerProvider = state != null ? state.getDiagnosticsProvider() : null;
         Object lockHolder = new Object();
-        if (tracerProvider == null ||
-            !tracerProvider.isEnabled()
-            || tracerProvider.shouldSampleOutOperation(pagedFluxOptions)) {
```

Comment:
why are we removing it from here?

Reply:
nit: If we are removing this then the `shouldSampleOutOperation` overload with `CosmosPageFluxOptions` remains unused and can be removed.  Also, a CosmosDiagnosticsContext can be constructed independently of tracer being enabled or not? Just curious on the coupling between the two concepts.

Reply:
If there is any diagnostics handler provided than the tracer is not disabled. When no tracing or no diagnostics are enabled then disabling the additional reactor overhead was intentional - and I would keep it for now (the overhead is >5 percent after all).

Reply:
So, yes - in theory you are right - but it is not possible without adding some overhead right now - and for the hotfix I think we should limit the scope as much as reasonable.